Build failing on 1.1.9

Hi there,

I'm using this library in my flutter app. I get a build error when trying to build apk:
```bash
❯ flutter build apk

e: file:///home/samoh/.pub-cache/hosted/pub.dev/flutter_audio_capture-1.1.9/android/src/main/kotlin/com/ymd/flutter_audio_capture/FlutterAudioCapturePlugin.kt:11:48 Unresolved reference: Registrar
e: file:///home/samoh/.pub-cache/hosted/pub.dev/flutter_audio_capture-1.1.9/android/src/main/kotlin/com/ymd/flutter_audio_capture/FlutterAudioCapturePlugin.kt:36:33 Unresolved reference: Registrar

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':flutter_audio_capture:compileReleaseKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,3 +60,8 @@
 ## 1.1.10
 
 - Resolved android run issue for Inconsistent JVM-target #36
+
+## 1.1.11
+
+- Removed deprecated `PluginRegistry.Registrar` usage in Android plugin to
+  resolve build errors on recent Flutter versions

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ Add this line to your pubspec.yaml file:
 
 ```
 dependencies:
-  flutter_audio_capture: ^1.1.10
+  flutter_audio_capture: ^1.1.11
 ```
 
 and execute

--- a/android/src/main/kotlin/com/ymd/flutter_audio_capture/FlutterAudioCapturePlugin.kt
+++ b/android/src/main/kotlin/com/ymd/flutter_audio_capture/FlutterAudioCapturePlugin.kt
@@ -8,7 +8,6 @@ import io.flutter.plugin.common.EventChannel
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
 
 /** FlutterAudioCapturePlugin */
 public class FlutterAudioCapturePlugin: FlutterPlugin, MethodCallHandler {
@@ -31,11 +30,6 @@ public class FlutterAudioCapturePlugin: FlutterPlugin, MethodCallHandler {
   // them functionally equivalent. Only one of onAttachedToEngine or registerWith will be called
   // depending on the user's project. onAttachedToEngine or registerWith must both be defined
   // in the same class.
-  companion object {
-    @JvmStatic
-    fun registerWith(registrar: Registrar) {
-    }
-  }
 
   override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
     when (call.method) {

--- a/example/lib/main.dart
+++ b/example/lib/main.dart
@@ -1,9 +1,19 @@
 import 'package:flutter/material.dart';
 import 'dart:async';
 
+import 'trip_form.dart';
 import 'package:flutter_audio_capture/flutter_audio_capture.dart';
 
-void main() => runApp(MyApp());
+void main() => runApp(const TripApp());
+class TripApp extends StatelessWidget {
+  const TripApp({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(home: TripFormPage());
+  }
+}
+
 
 class MyApp extends StatefulWidget {
   @override

--- a/example/lib/trip.dart
+++ b/example/lib/trip.dart
@@ -1,0 +1,14 @@
+class Trip {
+  final String name;
+  final DateTime startDate;
+  final DateTime endDate;
+
+  Trip({
+    required this.name,
+    required this.startDate,
+    required this.endDate,
+  });
+
+  @override
+  String toString() => 'Trip{name: $name, startDate: $startDate, endDate: $endDate}';
+}

--- a/example/lib/trip_form.dart
+++ b/example/lib/trip_form.dart
@@ -1,0 +1,153 @@
+
+import 'package:flutter/material.dart';
+
+import 'trip.dart';
+
+class TripFormPage extends StatefulWidget {
+  const TripFormPage({Key? key}) : super(key: key);
+
+  @override
+  _TripFormPageState createState() => _TripFormPageState();
+}
+
+class _TripFormPageState extends State<TripFormPage> {
+  final _formKey = GlobalKey<FormState>();
+  final TextEditingController _nameController = TextEditingController();
+  DateTime? _startDate;
+  DateTime? _endDate;
+
+  @override
+  void dispose() {
+    _nameController.dispose();
+    super.dispose();
+  }
+
+  Future<void> _pickStartDate() async {
+    final now = DateTime.now();
+    final picked = await showDatePicker(
+      context: context,
+      initialDate: _startDate ?? now,
+      firstDate: DateTime(now.year - 5),
+      lastDate: DateTime(now.year + 5),
+    );
+    if (picked != null) {
+      setState(() {
+        _startDate = picked;
+        if (_endDate != null && _endDate!.isBefore(_startDate!)) {
+          _endDate = null;
+        }
+      });
+    }
+  }
+
+  Future<void> _pickEndDate() async {
+    final now = DateTime.now();
+    final initial = _endDate ?? _startDate ?? now;
+    final picked = await showDatePicker(
+      context: context,
+      initialDate: initial,
+      firstDate: _startDate ?? DateTime(now.year - 5),
+      lastDate: DateTime(now.year + 5),
+    );
+    if (picked != null) {
+      setState(() {
+        _endDate = picked;
+      });
+    }
+  }
+
+  void _submit() {
+    if (_formKey.currentState?.validate() ?? false) {
+      final trip = Trip(
+        name: _nameController.text.trim(),
+        startDate: _startDate!,
+        endDate: _endDate!,
+      );
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text('Trip created: $trip')),
+      );
+    }
+  }
+
+  String? _validateName(String? value) {
+    if (value == null || value.trim().isEmpty) {
+      return 'Name is required';
+    }
+    return null;
+  }
+
+  String? _validateDates() {
+    if (_startDate == null || _endDate == null) {
+      return 'Please select both dates';
+    }
+    if (_endDate!.isBefore(_startDate!)) {
+      return 'End date must be after start date';
+    }
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Create Trip'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Form(
+          key: _formKey,
+          child: Column(
+            children: [
+              TextFormField(
+                controller: _nameController,
+                decoration: const InputDecoration(labelText: 'Name'),
+                validator: _validateName,
+              ),
+              const SizedBox(height: 16),
+              Row(
+                children: [
+                  Expanded(
+                    child: Text(_startDate == null
+                        ? 'Start date'
+                          : 'Start: ${_startDate!.toLocal().toString().split(' ')[0]}'),
+                  ),
+                  TextButton(
+                    onPressed: _pickStartDate,
+                    child: const Text('Select'),
+                  ),
+                ],
+              ),
+              Row(
+                children: [
+                  Expanded(
+                    child: Text(_endDate == null
+                        ? 'End date'
+                          : 'End: ${_endDate!.toLocal().toString().split(' ')[0]}'),
+                  ),
+                  TextButton(
+                    onPressed: _pickEndDate,
+                    child: const Text('Select'),
+                  ),
+                ],
+              ),
+              const SizedBox(height: 16),
+              Builder(
+                builder: (context) {
+                  final error = _validateDates();
+                  return error == null
+                      ? const SizedBox.shrink()
+                      : Text(error, style: TextStyle(color: Colors.red));
+                },
+              ),
+              const Spacer(),
+              ElevatedButton(
+                onPressed: _submit,
+                child: const Text('Create Trip'),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,6 +1,6 @@
 name: flutter_audio_capture
 description: Capture the audio buffer stream through microphone for iOS/Android.
-version: 1.1.10
+version: 1.1.11
 homepage: https://github.com/ysak-y/flutter_audio_capture
 
 environment:

